engine/src/java/com/runrev/android/Engine.java

```diff
@@ -2047,7 +2047,18 @@ public void showCamera()
 			return;
 		}
 
-		String t_path = m_temp_image_file.getPath();
+        // getPath() returns a path starting with /, e.g. /storage/emulated/0/Android/data/<app_id>/cache/..
+        // This path is appended to content://<app_id>.fileprovider/share/, resulting in an URI like:
+        //
+        // content://<app_id>.fileprovider//storage/emulated/0/Android/data/<app_id>/cache/..
+        //
+        // This is not a problem in general, but in _some_ devices (e.g. Huawei nova e3) this URI is
+        // normalized to:
+        //
+        // content://<app_id>.fileprovider/storage/emulated/0/Android/data/<app_id>/cache/..
+        //
+        // This results in the URI not being found. So use .substring(1) to remove the first slash
+		String t_path = m_temp_image_file.getPath().substring(1);
 
```

Comment:
Since the error is the produced URI, I think the most appropriate place to fix this would be in `FileProvider. getUriStringForFilePath(...)` where the URI is generated. You should also be checking whether the first char is '/' before doing this.

Reply:
Thanks @livecodeian, done :)